src/api_manager/proto/server_config.proto

```diff
@@ -231,4 +231,6 @@ message ApiServiceConfig {
 message Experimental {
   // Disable timed printouts of ESP status to the error log.
   bool disable_log_status = 1;
+  // Configure response to JSON translator option for JsonPrintOptions:
+  bool always_print_primitive_fields = 2;
```

Comment:
@lizan what is our plan?   
The plan to change service options field with selector is coming? is that right?   with that, users can set json transcoding field per method.
What is the plan for these flags in server config?   how do we combine them? 

Should we wait for the service protobuf change to get this feature in?

Reply:
@qiwzhang I don't believe this will interfere with Service/Method level configurations. This option is being applied as a service-wide "default" for the JsonPrintOptions as seen here:

https://github.com/vroomlabs/esp/blob/910e35ee875c5174b383d1940479ddc2281584b4/src/nginx/module.cc#L117

Presumably the existence of Service/Method configuration extensions would override this setting somewhere in the construction of the TranscoderFactory.

Reply:
In that case,   we should name it clearly, some things like "default_json_print_options".
There are other JsonPrintOptions fields,  I like to create a new proto Message as DefaultJsonPrintOptions at top level of server_config for them

Reply:
@qiwzhang I would agree, there is another option, add_whitespace; however, I would disagree with regards to the importance. While missing fields are fine for strong typed languages like C or C#, they are much more important to something like JavaScript. Whitespace is not a semantic change to the data and does not, at least to me, have the same importance.

We have been working to prioritize and resolve this for 3 months and have been very frustrated by the lack of response. We are inundated with defects caused by these missing fields and have intentionally avoided ESP and GRPC to avoid this issue at times. This "feature" was implemented by us to provide a minimal acceptable resolution to a problem that has cost hundreds of man hours and has had a serious adverse business impact here including both wasted time and lost sales. 

In the end, if implementing a 'default_json_print_options' or other task is required of us to move this forward we will of course do so. Just please come to an agreement on what the acceptance criteria is so that we can hit the mark and get this put down. We are extremely committed to closing this issue by any means possible as quickly as possible in order that we can stanch the bleeding of both time and money that this issue has been causing us.

Reply:
@qiwzhang It would only override the service config if this set to true, I don't think we need to rename it. Generally concept of service config is to override what is set in service config (e.g. service control / cloud trace server settings etc)